specs/governance/alligator.md

```diff
@@ -0,0 +1,390 @@
+
+When updating the subdelegation, the `subdelegate` function MUST override any previous subdelegation of the `msg.sender`
+to the `_delegatee`. Afterwards, this function MUST emit a `Subdelegation` event with the given function parameters.
+
+#### `subdelegateFromToken`
+
+Allows subdelegation of token voting power from an address (delegator) to another address (delegatee) with a specified
+subdelegation rule. This function is intended to be called by the `GovernanceToken` contract as part
+of its `delegateBySig` function. To ensure backwards compatibility in the `GovernanceToken`, the
+subdelegation rule MUST be 100% delegation, mimicking the behavior of the `GovernanceToken`'s `delegate`
+function.
+
+```solidity
+function subdelegateFromToken(address _delegator, address _delegatee, SubdelegationRule _rule) external
+```
+
+This function MUST check if the `_delegator` or `_delegatee` addresses have been migrated by checking the `migrated` mapping
+from its [storage](#storage). If either address has not been migrated, the `Alligator` MUST copy the delegation
+and checkpoint data from the token contract to its own state. After copying the data, the `Alligator` MUST update
+the `migrated` mapping to reflect that the address has been migrated.
+
+Before updating the subdelegation, the `subdelegateFromToken` function MUST check the validity of the subdelegation rule.
+
+When updating the subdelegation, the `subdelegateFromToken` function MUST override any previous subdelegation of the
+`_delegator` to the `_delegatee`. Afterwards, this function MUST emit a `Subdelegation` event with the given function
+parameters.
+
+#### `subdelegateBatched`
+
+Allows batch subdelegation of token voting power to multiple addresses with specified subdelegation rules. This
+function is intended to be called by users.
+
+```solidity
+function subdelegateBatched(address[] calldata _delegatees, SubdelegationRules[] calldata _rules) external
+```
+
+This function MUST check if the `msg.sender` has been migrated by checking the `migrated` mapping from its [storage](#storage).
+If it has not been migrated, the `Alligator` MUST copy the delegation and checkpoint data from the token contract
+to its own state. After copying the data, the `Alligator` MUST update the `migrated` mapping to reflect that the address
+has been migrated.
+
+This function MUST check that the length of `_delegatees` and `_rules` are equal. If the lengths are not equal, it MUST
+revert with an error.
+
+The `subdelegateBatched` function MUST iterate over each pair of `_delegatees` address and `_rules` subdelegation rule.
+At every iteration, the function MUST check the validity of the subdelegation rule, and migrate the  delegatee address
+if it has not been migrated. The function MUST then update the subdelegation.
+
+Afterwards, the `subdelegateBatched` function MUST emit a `Subdelegation` event with the given function parameters.
+
+#### `afterTokenTransfer`
```

Comment:
yea reiterating from the above and wonder if it's possible to emit events that reflect the updated voting power post subdelegation